macports/macports-ports
# py-scipy: update to 1.14.1

#### Description

PLEASE NOTE: this PR is on top of #26408 as there are several dependents in that PR that are needed for `py-scipy`. The NumPy PR will likely be merged first and then I will rebase this one. Please only focus on the changes related to SciPy here.

- update to latest upstream version
- drop py38, add py313 subport (to py-scipy as well as dependents)
- pin to 1.13.1 for py39 subport
- switch to PyPI
- add BLAS/LAPACK variants
- add myself as co-maintainer
- completely switch to 'meson' as build backend

Closes: https://trac.macports.org/ticket/70931
Closes: https://trac.macports.org/ticket/69538

@barracuda156 @aeiouaeiouaeiouaeiouaeiouaeiou same invitation to test building on older OSes. Please note that we have switched not completely to the `meson` backend; that is what upstream has done and thus not negotiable ;)


###### Type(s)



###### Tested on
macOS 15.1 24B83 x86_64
Xcode 16.1 16B40


###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?
